Source/SteveSloth/Public/MyEnemyAIIdleState.h

```diff
@@ -0,0 +1,36 @@
+/****************************************************************************************
+* Copyright: SteveSloth
+ * Name: Aidan MacLeod
+ * Script: MyEnemyAIIdleState.h
+ * Date: April 29. 2024
+ * Description: Idle State Class for all enemies to inherit from
+ * TODO:
+ * Known Bugs:
+ ****************************************************************************************/
+
+#pragma once
+
+#include "CoreMinimal.h"
+
+/**
+ * 
+ */
+
+UCLASS()
+class STEVESLOTH_API MyEnemyAIIdleState
+{
+
+private:
+
+	bool IsIdle;
+	bool HasDetectedPlayer;
+
+public:
+	
+	MyEnemyAIIdleState();
+	~MyEnemyAIIdleState();
+	
+private:
+
+	void DetectPlayer();
```

Comment:
Should be a lot more than a detectplayer function.
Needs to play an idle animation
Needs a count timer to go into patrol mode.
Needs to know where the enemy was spawned in, create a vector and add its initial location to it.
this way it knows where it started and where to go back to.

These are just some ideas.. Im sure there is more. but it is a start